wazo_dird/plugins/google_backend/services.py

```diff
@@ -36,39 +42,58 @@ def get_contacts(self, google_token, **list_params):
 
     def _fetch(self, google_token, term=None):
         headers = self.headers(google_token)
-        group_id = self._get_my_contacts_group_id(headers)
-        query_params = {'alt': 'json', 'max-results': 1000}
-        if term:
-            query_params['q'] = term
-        if group_id:
-            query_params['group'] = group_id
+        url = self.people_url
+        params = {
+            'personFields': self.person_fields,
+            'pageSize': 1000,
+        }
 
-        # TODO find a way to remove this verify = False
-        response = requests.get(
-            self.contacts_url, headers=headers, params=query_params, verify=False
+        if term:
+            url = self.search_url
+            params = {
+                'readMask': self.person_fields,
+                'pageSize': 30,
+            }
+
+            # empty request to 'warm' cache, recommended by Google
+            self._get_request(url, headers, params)
+            params['query'] = term
+
+        response = self._get_request(
+            url, headers, params, 'Fetched contacts from Google'
         )
-        if response.status_code != 200:
+        if not response:
             return []
 
-        logger.debug('Sucessfully fetched contacts from google')
-        logger.debug('Raw data: %s', response.text)
-        for contact in response.json().get('feed', {}).get('entry', []):
-            yield self.formatter.format(contact)
+        if term:
+            for contact in response.json().get('results', []):
+                yield self.formatter.format(contact.get('person', {}))
+        else:
+            for contact in response.json().get('connections', []):
+                yield self.formatter.format(contact)
 
-    def _get_my_contacts_group_id(self, headers):
-        query_params = {'alt': 'json'}
-        response = requests.get(
-            self.groups_url, headers=headers, params=query_params, verify=False
+    def _get_batch_of_contacts(self, headers, contact_ids):
+        params = [('personFields', self.person_fields)]
+        for contact_id in contact_ids:
+            params.append(('resourceNames', contact_id))
+
+        response = self._get_request(
+            self.batch_url, headers, params, 'Fetched batch of contacts from Google'
         )
+        return response.json().get('responses', [])
+
+    def _get_request(self, url, headers, params, debug_message=None):
+        # Requests have verify=False because the integration test mock servers do not have proper SSL
+        # Find a way to selectively turn off verification during testing only
+        response = requests.get(url, headers=headers, params=params, verify=False)
         if response.status_code != 200:
+            logger.debug('Get Request Unsuccessful: %s' % str(response.status_code))
```

Comment:
You should not use `%` to format strings. The `%s` also does the `str()` on the parameter. This should be
```python
logger.debug('Get Request Unsuccessful: %s', response.status_code)
```
Same thing for the other logger.debug.

You can safely forget about the `%` operator for string formating ;)